mongodbatlas/privatelink_endpoints_adl.go

```diff
@@ -0,0 +1,158 @@
+// Copyright 2021 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mongodbatlas
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+)
+
+const (
+	privateLinkEndpointsADLPath = "api/atlas/v1.0/groups/%s/privateNetworkSettings/endpointIds"
+)
+
+// PrivateLinkEndpointsADLService is an interface for interfacing with the Private Link Endpoints ADL
+// of the MongoDB Atlas API datalake and online archive.
+//
+// See more: https://docs.atlas.mongodb.com/reference/api/private-endpoints/
```

Comment:
this link is not correct which leads me to a major issue here.

Given this statement about the go client lib README

> The services of a client divide the API into logical chunks and correspond to
the structure of the Atlas API documentation at
https://docs.atlas.mongodb.com/api/.

You have two options

Add the methods to `OnlineArchiveService` (https://docs.atlas.mongodb.com/reference/api/online-archive/)
Or add the methods to `DataLakeService` (https://docs.mongodb.com/datalake/reference/api/datalakes-api/)

Even adding to both may be the correct thing

Reply:
my bad, I copy from private_link but I forgot to change this

Reply:
already changed

Reply:
Thanks for updating the link but that doesn't solve my main comment, let's not add a new service, let's add the method to `OnlineArchiveService`, `DataLakeService` or both

Reply:
Oh I understand, but now my concern is, it is okay to use `DataLakeService` when you are using `OnlineArchiveService` and want to create a private_link?

Reply:
What happen if the API documentation for the `private_link_endpoint` for this gets a new section? you will need to "refactor" this services?

Reply:
> is it okay to use DataLakeService when you are using OnlineArchiveService and want to create a private_link?

I say yes but up to you, remember one of the options is to add to both 

> What happen if the API documentation for the private_link_endpoint for this gets a new section?

We have a "no breaking changes" approach on the client unless strictly necessary, if the docs are moved around we should not move code around, we'll live with the choices made with what we knew today

Reply:
Ready! I added this new methods on both datalake and online archive services